packages/core/src/entrypoint/index.ts

```diff
@@ -0,0 +1,189 @@
+};
+
+export function getEntryPoint<
+  TEntryPointVersion extends EntryPointVersion = DefaultEntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[TEntryPointVersion];
+
+export function getEntryPoint<
+  TEntryPointVersion extends DefaultEntryPointVersion = DefaultEntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options?: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[TEntryPointVersion];
+
+export function getEntryPoint<TChain extends Chain = Chain>(
+  chain: TChain,
+  options?: GetEntryPointOptions<DefaultEntryPointVersion>
+): EntryPointDefRegistry<TChain>[DefaultEntryPointVersion];
+
+export function getEntryPoint<
+  TEntryPointVersion extends EntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[EntryPointVersion] {
+  const { version, addressOverride } = options ?? {
+    version: defaultEntryPointVersion,
+  };
+  const entryPoint = entryPointRegistry[version ?? defaultEntryPointVersion];
+  const address = addressOverride || entryPoint.address[chain.id];
+  if (!address) {
+    throw new EntryPointNotFoundError(chain, version);
+  }
+
+  if (version === "0.6.0") {
+    const v6 = entryPoint as SupportedEntryPoint<
+      "0.6.0",
+      TChain,
+      typeof EntryPointAbi_v6
+    >;
+    return {
+      version: v6.version,
+      address,
+      chain,
+      abi: v6.abi,
+      getUserOperationHash: (r) =>
+        v6.getUserOperationHash(r, address, chain.id),
+      packUserOperation: v6.packUserOperation,
+      isUserOpVersion: v6.isUserOpVersion,
+      coerce: (userOperation, throwable) =>
+        coerceEntryPointVersion(v6, userOperation, throwable),
+    } as EntryPointDef<"0.6.0", TChain, typeof EntryPointAbi_v6>;
+  } else if (version === "0.7.0") {
+    const v7 = entryPoint as SupportedEntryPoint<
+      "0.7.0",
+      TChain,
+      typeof EntryPointAbi_v7
+    >;
+    return {
+      version: v7.version,
+      address,
+      chain,
+      abi: v7.abi,
+      getUserOperationHash: (r) =>
+        v7.getUserOperationHash(r, address, chain.id),
+      packUserOperation: v7.packUserOperation,
+      isUserOpVersion: v7.isUserOpVersion,
+      coerce: (userOperation, throwable) =>
+        coerceEntryPointVersion(v7, userOperation, throwable),
+    } as EntryPointDef<"0.7.0", TChain, typeof EntryPointAbi_v7>;
+  }
+
+  throw new EntryPointNotFoundError(chain, version);
+}
+
+// // =================================================================================
+// // TODO: Add tests for the following cases
+
+// const chain: Chain = mainnet;
+
+// // 1. left and right hand side version type should match, as well as the param
+
+// // === Error ===
+// const ep1_a: EntryPointDef<"0.6.0"> = getEntryPoint(chain, {
```

Comment:
You may know this already, but if you're trying to write tests that certain expressions do _not_ typecheck, you can use `// @ts-expect-error` directives ([docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments)).

Reply:
follow up PR moves this to actual unit test